### DISCUSSION TITLE
Catalog.list_items failed due to missing parameter

### DISCUSSION THREAD

**Discussion by icexxxxxx** [saleweaver/python-amazon-sp-api#87] [Q&A] [answered]
Hi all,

I tried to fetch some data with Catalog.list_items(/catalog/v0/items) following the official docs, which the only required param is MarketplaceId.

However, the error occurs always:

```
[{'code': 'InvalidInput', 'message': 'Required parameter IdType not found', 'details': ''}]
```

I wonder if anyone met this before and what the IdType is.  I found hc's answer in  https://sellercentral.amazon.com/forums/t/need-help-adding-string-to-idlist/187687/3, but it doesn't work either.

Nothing else is responded even if I write like this:

```python
Catalog(credentials=credentials).list_items(
        IdType='ASIN',
        MarketplaceId='ATVPDKIKX0DER',
)
```

I'm new to the Amazon Seller API, thanks for any help!🤦‍♀️

**Comment by yberezkin** [comment, marked as answer]
You can't use Catalog with ASIN. Its primary use case is to find ASINS by using another type of identifier.

Quote from the [documentation](https://github.com/amzn/selling-partner-api-docs/blob/main/references/catalog-items-api/catalogItemsV0.md):


Type | Name | Description | Schema
Query | MarketplaceId  required | A marketplace identifier. Specifies the marketplace for which items are returned. | string
Query | Query  optional | Keyword(s) to use to search for items in the catalog. Example: 'harry potter books'. | string
Query | QueryContextId  optional | An identifier for the context within which the given search will be performed. A marketplace might provide mechanisms for constraining a search to a subset of potential items. For example, the retail marketplace allows queries to be constrained to a specific category. The QueryContextId parameter specifies such a subset. If it is omitted, the search will be performed using the default context for the marketplace, which will typically contain the largest set of items. | string
Query | SellerSKU  optional | Used to identify an item in the given marketplace. SellerSKU is qualified by the seller's SellerId, which is included with every operation that you submit. | string
Query | UPC  optional | A 12-digit bar code used for retail packaging. | string
Query | EAN  optional | A European article number that uniquely identifies the catalog item, manufacturer, and its attributes. | string
Query | ISBN  optional | The unique commercial book identifier used to identify books internationally. | string
Query | JAN  optional | A Japanese article number that uniquely identifies the product, manufacturer, and its attributes. | string


Example of calls:
```python
resp = Catalog(credentials=credentials).list_items(
    MarketplaceId=sp_api.base.Marketplaces.US.marketplace_id,
    Query='Python Coding')
```
That's not very efficient in Amazon to use regular Query search.
But here an example with using items IDs like UPC:

```python
resp = Catalog(credentials=credentials).list_items(
        MarketplaceId=sp_api.base.Marketplaces.US.marketplace_id,
        UPC='885909918164')
```

**Reply by icexxxxxx** [reply]
Much appreciate !!!

**Reply by yberezkin** [reply]
No problem! 👍

**Reply by ghost** [reply]
@yberezkin 
Is this UPC='885909918164' is dummy one. I'm asking this because when i'm using GET https://sellingpartnerapi-na.amazon.com/catalog/v0/items?MarketplaceId=ATVPDKIKX0DER&UPC=885909918164 , it should hitting production box? Also, pls let me know how we call sandbox here. What changes i need to do in the above URL? Pls suggest.
